docs/candidate-app/password-validation.md

```diff
@@ -0,0 +1,101 @@
+# Password Validation
+
+## Frontend
+
+Setting the password is done in two components:
+
+- [`PasswordSetPage.svelte`](/frontend/src/routes/candidate/register/PasswordSetPage.svelte) (new user)
+- [`PasswordResetPage.svelte`](/frontend/src/routes/candidate/password-reset/PasswordResetPage.svelte) (forgotten password)
+
+The password is validated using [`passwordValidation.ts`](/frontend/src/lib/utils/passwordValidation.ts).
+
+The file provides two main functions for validating a password
+
+- `validatePassword`  
+  Returns a boolean indicating whether the password is valid
+- `validatePasswordDetails`  
+  Returns both the password validity status boolean as well as details and state for each password requirement. These details are used for the validation UI.
+
+On both password set pages, a password validation UI is shown to the user.
+This functionality is provided by the [`PasswordValidator.svelte`](/frontend/src/lib/components/passwordValidator/PasswordValidator.svelte) component.
+The component provides real-time password validation checks and shows the state of each password requirement.
+More details on how the different requirements are shown to the user can be found in the documentation of the component.
+
+The UI validation is done using debounced validation, i.e. the validation status is checked and updated once the user stops typing for a moment.
+
+The password validation UI provides a variable indicating the password validation status, which can be used to enable/disable the password submit button. This way frontend allows the password to be submitted only if it is valid.
+
+However, due to the use of debounced validation, the validity of the password is also checked by the password set page itself before sending the request to the backend.
+
+If the validity check passes, a POST request is sent to the backend to either
+
+- `api/auth/candidate/register`
+- `api/auth/reset-password`
+
+## Backend
+
+Before accepting the password, the backend also validates the password.
+This is done in the [`candidate.ts`](/backend/vaa-strapi/src/extensions/users-permissions/controllers/candidate.ts) file using the same `validatePassword` function from [`passwordValidationCopy.ts`](/backend/vaa-strapi/src/util/passwordValidationCopy.ts).
+Currently, the backend uses a copy of the original validation file.
```

Comment:
Can we also add a note here regarding what files specifically implement which endpoint? Something like:
- candidate.ts implements the /api/auth/candidate/register endpoint using the registration key
- /api/auth/reset-password is fully handled by Strapi's users-permissions plugin (currently this isn't mentioned at all so it may confuse the developer that this is something managed by us)